playbooks/tasks/portal-docker-services-check-running.yml

```diff
@@ -0,0 +1,119 @@
+---
+# Check docker services are running (not restarting)
+# Inputs:
+# - docker_containers_check_list: List of docker container names to check
+# - timeout_secs: Total timeout in seconds to wait for containers running
+#     successfully, not restarting (expected to be 120+ seconds)
+
+- name: Assert that inputs are defined correctly
+  ansible.builtin.assert:
+    that:
+      - docker_containers_check_list is defined
+      - docker_containers_check_list | length > 0
+      - docker_containers_check_timeout_secs is defined
+      - docker_containers_check_timeout_secs > 0
+    fail_msg: |
+      Variable docker_containers_check_list must be defined
+      and it must be a list containing at least 1 docker service.
+
+- name: Set main start timestamp once
+  set_fact:
+    # We want to get local time using lookup/pipe
+    check_start_epoch: "{{ lookup('ansible.builtin.pipe', 'date +%s') }}"
+  when: >-
+    check_start_epoch is not defined
+    or check_start_epoch == 0
+
+# We want to get current check timestamp from a remote host, because docker
+# container info StartedAt timestamp is also from remote host, so we can't use
+# lookup/pipe.
+
+- name: Get current check epoch timestamp from remote host
+  ansible.builtin.command: date +%s
+  register: date_result
+
+- name: Set current check epoch timestamp from remote host
+  set_fact:
+    current_check_start_epoch: "{{ date_result.stdout }}"
+
+- name: Check docker containers status
+  community.docker.docker_container_info:
+    name: "{{ item }}"
+  register: docker_info_result
+  loop: "{{ docker_containers_check_list }}"
+
+- name: Reset docker services are running successfully flag
+  set_fact:
+    docker_services_running_ok: True
+
+- name: Set docker services are running successfully flag
+  vars:
+    min_required_non_restarting_time_in_secs: 65
+  set_fact:
+    # Check that container exists, is running and is not restarting for more
+    # more than 1 minute. We need to wait more than 1 minute, because container
+    # can seem running ok (not restating) and restart after 1 minute.
+    # Time check condition:
+    #   (current_check_start_epoch in UTF in secs) - (container started at)
+    #   > min time required
+    docker_services_running_ok: >-
+      {{
+        docker_services_running_ok
+        and item.exists
+        and item.container.State.Running
+        and not item.container.State.Restarting
+        and
+          current_check_start_epoch | int
```

Comment:
Shouldn't this be the local `date` you mentioned?

Reply:
No, because it compares `container.State.StartedAt` which is remote time from ansible wrapper over remote `docker inspect {container-name}`, so `current_check_start_epoch` should be also remote time.

Reply:
Oh, that's `current_check_start_epoch` and not `check_start_epoch`! My mistake.

Reply:
... `current_check_start_epoch` is using remote `date` via `ansible.builtin.command` run on remote host.